### ISSUE TITLE
Questions about integrations and scope of `dvclive`

### ISSUE THREAD

**Issue by daavoo** [iterative/dvclive#70]
Hi there!

First of all, thank you for your open source efforts.

I have been using `dvc` for a while and, as we are migrating our workflows to the new version `2.X`, we are considering to integrate `dvclive` and willing to contribute. After reading the [documentation](https://dvc.org/doc/dvclive) along with #5 and #66 I am a little confused about the scope of the project and how to proceed with the integration. 

We mainly use [mmcv](https://github.com/open-mmlab/mmcv) and (to a lesser extent) [pytorch-lightning ](https://github.com/PyTorchLightning/pytorch-lightning) as training frameworks. In both those frameworks the usual procedure for adding support to a new metric tracking tool would be to send a P.R. directly to those repositories adding a new `Logger` ([pytorch-lightning](https://github.com/PyTorchLightning/pytorch-lightning/tree/master/pytorch_lightning/loggers) / [mmcv](https://github.com/open-mmlab/mmcv/tree/master/mmcv/runner/hooks/logger)). This leads me to the next question:

1) Would be better to send a P.R. to those frameworks or to this repository (as I understand reading #5) ?

In addition, as my understanding of the official [documentation](https://dvc.org/doc/dvclive) and [release post](https://dvc.org/blog/dvc-2-0-release) does not match the direction described in #66:

2) Should we consider `dvclive`, in the short term,  as an *alternative*, *companion* or *wrapper* to other metric tracking tools we might be currently using (i.e. [MLflow Tracking ](https://mlflow.org/docs/latest/tracking.html))

Thanks in advance

**Comment by dmpetrov** [comment]
Hi @daavoo, that's a great question!

You are right - `dvclive` is an alternative to MlFlow tracking and you can take a look at MlFlow integration as an example. If there is a lack of some functionality on dvclive side - new features can be implemented here.

Regarding MMCV - I have not used it, unfortunately, (is it like PyTorch, Keras libs?). Based on my understanding the implementation might be needed on both sides.

Yes, we should be more direct about this on the main page https://github.com/iterative/dvclive.

We are eager to help you with the integrations!

**Comment by dmpetrov** [comment]
I'm updating readme docs. I hope it will help with the product positioning.

PR: https://github.com/iterative/dvclive/pull/71= Feedback is welcome!

**Comment by daavoo** [comment]
> Hi @daavoo, that's a great question!
> 
> You are right - `dvclive` is an alternative to MlFlow tracking and you can take a look at MlFlow integration as an example. If there is a lack of some functionality on dvclive side - new features can be implemented here.

Thanks for the clarification @dmpetrov!

> Regarding MMCV - I have not used it, unfortunately, (is it like PyTorch, Keras libs?). Based on my understanding the implementation might be needed on both sides.

mmcv It's more like an opinionated higher level framework (it *uses* PyTorch) focused on Computer Vision, and it's the core behind other task-specific libraries like [mmdetection](https://github.com/open-mmlab/mmdetection) or [mmsegmentation](https://github.com/open-mmlab/mmsegmentation).

I think that whether the `Logger` implementation should live inside `dvclive` or the framework itself would be more conditioned by product/community considerations than implementation needs. I also feel like it depends on the scope of each framework.

Taking a popular `dvclive` alternative, `wandb`, as example: For "higher level" frameworks (i.e. `fastai`, `pytorch-lighting`) the `Logger` implementation lives in the framework repository, and it is referenced in the documentation:

- https://docs.wandb.ai/integrations/fastai
- https://docs.wandb.ai/integrations/lightning

However, for "low level" frameworks like Keras, PyTorch or XGBoost, they provide and maintain a reference implementation inside the `wandb` library:

- https://docs.wandb.ai/integrations/keras
- https://docs.wandb.ai/integrations/pytorch
- https://docs.wandb.ai/integrations/xgboost

From both a `dvc` user and member of the open source community perspectives, i like this approach as it creates connections between different parts of my usual workflow.

From a contributor/maintainer perspective (i.e. I contribute to `mmcv`) I see the potentials benefits of maintaining the `Logger` implementation inside each framework. A concrete example: I don't use the `wandb` logger from `mmcv`, but because the `Logger` lives inside the framework, the `wandb` logger got features and bugfixes as part of a refactor that [I contributed ](https://github.com/open-mmlab/mmcv/pull/605) to the base `Logger` module. I believe there are many more similar examples in other frameworks.

**Comment by daavoo** [comment]
> I'm updating readme docs. I hope it will help with the product positioning.
> 
> PR: #71 Feedback is welcome!

Nice! I feel like the new description is much more enlightening
